set.go

```diff
@@ -0,0 +1,409 @@
+package crdt
+
+import (
+	"bytes"
+	"encoding/binary"
+	"errors"
+	"strings"
+
+	pb "github.com/ipfs/go-ds-crdt/pb"
+
+	ds "github.com/ipfs/go-datastore"
+	query "github.com/ipfs/go-datastore/query"
+)
+
+var (
+	elemsNs        = "s" // elements namespace
+	tombsNs        = "t" // tombstones namespace
+	valueSuffix    = "v"
+	prioritySuffix = "p"
+)
+
+// set implements an Add-Wins Observed-Remove Set using delta-CRDTs
+// (https://arxiv.org/abs/1410.2803) and backing all the data in a
+// go-datastore. It is fully agnostic to MerkleCRDTs or the delta distribution
+// layer.  It chooses the Value with most priority for a Key as the current
+// Value. When two values have the same priority, it chooses by alphabetically
+// sorting their unique IDs alphabetically.
+type set struct {
+	store     ds.Datastore
+	namespace ds.Key
+}
+
+func newCRDTSet(d ds.Datastore, namespace ds.Key) *set {
+	s := &set{
+		namespace: namespace,
+		store:     d,
+	}
+	return s
+}
+
+// Add returns a new delta-set adding the given elements.
+func (s *set) Add(key string, value []byte) *pb.Delta {
+	return &pb.Delta{
+		Elements: []*pb.Element{
+			&pb.Element{
+				Key:   key,
+				Value: value,
+			},
+		},
+		Tombstones: nil,
```

Comment:
Seems the comment is old. Updated it.

> This could not be a method of set and would still work, so why?

Because it is way clearer that you are getting the delta to add to a set, just like `Rmv`, regardless of whether the set is needed internally or not. 

> Looking at the code it doesn't seem like it is not adding, but rather it returns a new pb.Delta

This is how the Add operation is defined in delta-crdts. Deltas need to be merged later.